x-pack/plugins/cloud/server/config.ts

```diff
@@ -22,6 +22,10 @@ const configSchema = schema.object({
   apm: schema.maybe(apmConfigSchema),
   base_url: schema.maybe(schema.string()),
   cname: schema.maybe(schema.string()),
+  // Cloud service provider
+  csp: schema.maybe(
+    schema.oneOf([schema.literal('aws'), schema.literal('gcp'), schema.literal('azure')])
```

Comment:
NIT: for the same reason we're `maybe`-ing all cloud config entries, shall we also accept unknown/arbitrary values for the cloud service provider, in case of other ones get added later?

Reply:
I agree, let's add a schema.string() to that list